static/modes/glsl-mode.ts

```diff
@@ -0,0 +1,745 @@
+        'float16_t',
+        'float32_t',
+        'float64_t',
+        'int64_t',
+        'uint64_t',
+        'int32_t',
+        'uint32_t',
+        'int16_t',
+        'uint16_t',
+        'int8_t',
+        'uint8_t',
+
+        // GLSL.std.450.pdf
+        'radians',
+        'degrees',
+        'sin',
+        'cos',
+        'tan',
+        'asin',
+        'acos',
+        'atan',
+        'sinh',
+        'cosh',
+        'tanh',
+        'asinh',
+        'acosh',
+        'atanh',
+        'pow',
+        'exp',
+        'log',
+        'exp2',
+        'log2',
+        'sqrt',
+        'inversesqrt',
+        'abs',
+        'sign',
+        'floor',
+        'trunc',
+        'round',
+        'roundEven',
+        'ceil',
+        'fract',
+        'mod',
+        'modf',
+        'min',
+        'max',
+        'clamp',
+        'mix',
+        'step',
+        'smoothstep',
+        'isnan',
+        'isinf',
+        'floatBitsToInt',
+        'floatBitsToUint',
+        'intBitsToFloat',
+        'uintBitsToFloat',
+        'fma',
+        'frexp',
+        'ldexp',
+        'packUnorm2x16',
+        'packSnorm2x16',
+        'packUnorm4x8',
+        'packSnorm4x8',
+        'unpackUnorm2x16',
+        'unpackSnorm2x16',
+        'unpackUnorm4x8',
+        'unpackSnorm4x8',
+        'packHalf2x16',
+        'unpackHalf2x16',
+        'packDouble2x32',
+        'unpackDouble2x32',
+        'length',
+        'distance',
+        'dot',
+        'cross',
+        'normalize',
+        'faceforward',
+        'reflect',
+        'refract',
+        'matrixCompMult',
+        'outerProduct',
+        'transpose',
+        'determinant',
+        'inverse',
+        'lessThan',
+        'lessThanEqual',
+        'greaterThan',
+        'greaterThanEqual',
+        'equal',
+        'notEqual',
+        'any',
+        'all',
+        'not',
+        'uaddCarry',
+        'usubBorrow',
+        'umulExtended',
+        'bitfieldExtract',
+        'bitfieldInsert',
+        'bitfieldReverse',
+        'bitCount',
+        'findLSB',
+        'findMSB',
+        'dFdx',
+        'dFdxFine',
+        'dFdxCoarse',
+        'dFdy',
+        'dFdyFine',
+        'dFdyCoarse',
+        'fwidth',
+        'fwidthFine',
+        'fwidthCoarse',
+        'anyInvocation',
+        'allInvocations',
+        'allInvocationsEqual',
+
+        // misc samplers
+        'isamplerCube',
+        'isamplerCubeArray',
+        'usamplerCube',
+        'usamplerCubeArray',
+        'isampler3D',
+        'usampler3D',
+        'sampler3D',
+        'sampler2DRect',
+        'sampler2DRectShadow',
+        'isampler2DRect',
+        'usampler2DRect',
+        'sampler2DMS',
+        'isampler2DMS',
+        'usampler2DMS',
+        'sampler2DMSArray',
+        'isampler2DMSArray',
+        'usampler2DMSArray',
+        'samplerBuffer',
+        'isamplerBuffer',
+        'usamplerBuffer',
+        'samplerShadow',
+
+        // misc texture
```

Comment:
are these really keywords? These seem like built in functions?

Reply:
well I guess not sure if these counted, like they are reserved names in the GLSL language, just they are functions... happy to remove or group together

Reply:
(I also was kinda surprised by how many keywords GLSL has accumulated, I guess my one counter-argument is for HLSL they also have functions in the keyword list, so it is at least consistent

Reply:
I'm cool with that then, if you're following what HLSL is doing.